src/python.rs

```diff
@@ -189,6 +189,24 @@ impl<'p> Python<'p> {
         GILGuard::acquire()
     }
 
+    /// Acquires the [GILGuard] without performing state checking.
+    ///
+    /// # Safety
+    ///
+    /// This bypasses checking that [Python::acquire_gil] would normally perform, such
+    /// as ensuring the Python interpreter is fully initialized. If you call this
+    /// from a process where the Python interpreter isn't in a "good" state, the
+    /// process may crash.
+    ///
+    /// One special case where calling this function over [Python::acquire_gil] is
+    /// justified is during multi-phase interpreter initialization. If the interpreter
+    /// is configured for multi-phase initialization, it is safe to call this function
+    /// between `Py_InitializeFromConfig()` and `_Py_InitializeMain()`.
```

Comment:
I can't give a full accounting of when this API is and is not safe to use because I'm unsure and the answer likely depends on Python interpreter implementation details.

I do know it seems to be safe between `Py_InitializeFromConfig` and `_Py_InitializeMain`.

Not all Python interpreter features are available before `_Py_InitializeMain`. However, many are. And so far all PyO3 functionality that PyOxidizer uses before calling `_Py_InitializeMain` seems to work. (But I haven't ported everything.) For an idea of what doesn't work before `_Py_InitializeMain` is called, you'll have to look at the CPython source code. This code path has gotten refactored heavily in Python 3.8-3.10 and TBH I'm unsure of the current state of it, so I don't want to make any false claims. The last time I looked at things, initialization stopped just before importing any .py modules on the filesystem. So you have builtin extension modules along with other core features from the interpreter. But nothing from .py modules since none are loaded yet.